fetch.js

```diff
@@ -85,6 +85,20 @@ export function putJSON(url, body, callback) {
     return fetchJSON(url, 'PUT', 'include', body, callback);
 }
 
+/**
+ * Download and parse a remote JSON endpoint via PATCH. credentials
+ * are included automatically
+ *
+ * @param {string} url
+ * @param {string} body
+ * @param {function} callback
+ *
+ * @returns {Promise}
+ */
+export function patchJSON(url, body, callback) {
+    return fetchJSON(url, 'PATCH', 'include', body, callback);
```

Comment:
I'm a little confused why these functions exist. As I understand it `fetchJSON` is a small wrapper around `window.fetch` and it simplifies the API so the consumer doesn't have to do the `await res.json()` dance. In most cases it is just moving a parameter into the function name `patchJSON` instead of `fetchJSON('PATCH')`. Then for every HTTP method there is a function which I import, but I could also directly use `fetchJSON` for all of them.

Wouldn't this be enough:

```js
function fetchJSON(url, options) {
    Object.assign(options, { mode: 'cors' });
    return window
        .fetch(url, options)
        .then(res => {
            if (res.ok) return res.json();
            throw new Error(res.statusText);
        })
        .catch(err => console.error(err));
}
```

I guess that's how these functions grew over time. I'm curious if I'm missing something.

Reply:
We've decided to replace `fetchJSON` and its friends with a new API wrapper.